### ISSUE TITLE
[BUG] Replica can run cross-slot MULTI/EXEC, upon module hooks on primary

### ISSUE THREAD

**Issue by kyle-yh-kim** [madolson/placeholderkv#832]
**Describe the bug**

The following tcl test case can be found within our `unit/moduleapi/cluster` test class. There are three commands in play; `del` (original command from Primary), `keyspace.incr_dels` and `lpush` (from module keyspace event callback).

In the primary, these are not executed under `multi/exec`, however they are replicated as a transaction under replica. This can cause the replica to execute cross-slot transaction.

```tcl
 the {lpush before_deleted count_dels_{4oi}} is a post notification job registered when 'count_dels_{4oi}' was removed
assert_replication_stream $repl {
    {multi}
    {del count_dels_{4oi}}
    {keyspace.incr_dels}
    {lpush before_deleted count_dels_{4oi}}
    {exec}
}
```

https://github.com/valkey-io/valkey/blob/unstable/tests/unit/moduleapi/cluster.tcl#L210-L217

**To reproduce**

Steps to reproduce the behavior and/or a minimal code sample.

**Expected behavior**

A description of what you expected to happen.

**Additional information**

Any additional information that is relevant to the problem.

**Comment by zuiderkwast** [comment]
Does replication fail?

**Comment by madolson** [comment]
> Does replication fail?

No. Replicas don't call getNodeByQuery and just blindly trust the commands they are receiving from their primary.
